src/storage/iceberg_delete.cpp

```diff
@@ -0,0 +1,358 @@
+#include "storage/iceberg_delete.hpp"
+#include "storage/irc_catalog.hpp"
+#include "storage/irc_transaction.hpp"
+#include "storage/irc_table_entry.hpp"
+#include "storage/iceberg_table_information.hpp"
+#include "iceberg_multi_file_reader.hpp"
+#include "iceberg_multi_file_list.hpp"
+
+#include "duckdb/planner/expression/bound_reference_expression.hpp"
+#include "duckdb/catalog/catalog_entry/copy_function_catalog_entry.hpp"
+#include "duckdb/execution/operator/scan/physical_table_scan.hpp"
+#include "duckdb/execution/operator/persistent/physical_copy_to_file.hpp"
+#include "duckdb/execution/physical_plan_generator.hpp"
+#include "duckdb/execution/physical_operator.hpp"
+#include "duckdb/function/copy_function.hpp"
+#include "duckdb/common/types/uuid.hpp"
+#include "duckdb/common/multi_file/multi_file_reader.hpp"
+#include "fmt/format.h"
+
+namespace duckdb {
+class IcebergDeleteLocalState;
+class IcebergDeleteGlobalState;
+class ICTableEntry;
+
+IcebergDelete::IcebergDelete(PhysicalPlan &physical_plan, ICTableEntry &table, PhysicalOperator &child,
+                             shared_ptr<IcebergDeleteMap> delete_map_p, vector<idx_t> row_id_indexes)
+    : PhysicalOperator(physical_plan, PhysicalOperatorType::EXTENSION, {LogicalType::BIGINT}, 1), table(table),
+      delete_map(std::move(delete_map_p)), row_id_indexes(std::move(row_id_indexes)) {
+	children.push_back(child);
+}
+
+unique_ptr<GlobalSinkState> IcebergDelete::GetGlobalSinkState(ClientContext &context) const {
+	return make_uniq<IcebergDeleteGlobalState>();
+}
+
+unique_ptr<LocalSinkState> IcebergDelete::GetLocalSinkState(ExecutionContext &context) const {
+	return make_uniq<IcebergDeleteLocalState>();
+}
+
+//===--------------------------------------------------------------------===//
+// Sink
+//===--------------------------------------------------------------------===//
+SinkResultType IcebergDelete::Sink(ExecutionContext &context, DataChunk &chunk, OperatorSinkInput &input) const {
+	auto &global_state = input.global_state.Cast<IcebergDeleteGlobalState>();
+	auto &local_state = input.local_state.Cast<IcebergDeleteLocalState>();
+
+	auto &file_name_vector = chunk.data[row_id_indexes[0]];
```

Comment:
This is being accessed with the selection vector from `file_row_number`, which aren't guaranteed to match
You should create a UnifiedVectorFormat for this Vector as well

Reply:
I'm not sure what you mean here, the file_row_number and file_name_vector are both converted to the UnifiedVectorFormat in lines after this, into `row_data` and `file_name_vdata`. Once those are created `file_name_vector` and `file_row_number` are no longer used